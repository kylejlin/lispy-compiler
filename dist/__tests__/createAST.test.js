'use strict';

var _createAST = require('../createAST');

var _createAST2 = _interopRequireDefault(_createAST);

var _tokenize = require('../tokenize');

var _tokenize2 = _interopRequireDefault(_tokenize);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var code = '\n(pickRandom "foo" 9 (sum 1 2 3))\n';

var tokens = (0, _tokenize2.default)(code);
var ast = (0, _createAST2.default)(tokens);

test('it generates the correct AST', function () {
  expect(ast).toMatchSnapshot();
});