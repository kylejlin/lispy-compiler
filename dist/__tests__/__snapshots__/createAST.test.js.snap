// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it generates the correct AST 1`] = `
Object {
  "body": Object {
    "arguments": Array [
      Object {
        "type": "StringLiteral",
        "value": "foo",
      },
      Object {
        "type": "NumberLiteral",
        "value": "9",
      },
      Object {
        "arguments": Array [
          Object {
            "type": "NumberLiteral",
            "value": "1",
          },
          Object {
            "type": "NumberLiteral",
            "value": "2",
          },
          Object {
            "type": "NumberLiteral",
            "value": "3",
          },
        ],
        "callee": "sum",
        "type": "CallExpression",
      },
    ],
    "callee": "pickRandom",
    "type": "CallExpression",
  },
  "type": "Program",
}
`;
